src/homework01/Tasks.kt

```diff
@@ -1,13 +1,228 @@
 package homework01
 
-fun binarySearch(array: List<Int>, f: Int): Int = 0
+/**
+ * -1 is added when value not found
+ * to avoid ambiguous results when value
+ * could've been inserted to 0th index
+ *
+ * @see binarySearchRecursion
+ */
+fun linearSearch(array: List<Int>, element: Int, start: Int, end: Int, compare: (Int, Int) -> Int): Int {
+    for (i in start..end) {
+        if (compare(array[i], element) == 0) {
+            return i
+        }
+        if (compare(array[i], element) > 0) {
+            return -i - 1
+        }
+    }
+    return -end - 1
+}
 
-fun howManyNumbers(array: List<Int>, l: Int, r: Int): Int = 0
+/**
+ * start and end are assumed to be in [0..array.size]
+ * also assumed that start <= end
+ * array is assumed to be sorted in ascending order according to comparator
```

Comment:
It's cool, that you assume it, but it'll be better to check using `assert`.
In statement I write, that array is sorted, so you don't need to check it.
Also, it's good, that you are using comparator, we will talk about it latter, but currently we work only with number, so you can use regular comparison.

Reply:
I'd also suggest, if you want to generalize binarySearch for any objects, you could make [Generic function](https://kotlinlang.org/docs/reference/generics.html#generic-functions) with either type implementing Comparable, or take objects and comparator on objects.